.github/workflows/main.yml

```diff
@@ -1,35 +1,58 @@
 name: Actions
 
-on: [push, pull_request]
+on: [ push, pull_request ]
 
 jobs:
-  test:
-    name: "Test"
+  validation:
+    name: "Gradle wrapper validation"
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: gradle/wrapper-validation-action@v1
+  build:
+    name: "Build"
+    needs: [ validation ]
     strategy:
       matrix:
-        # 17 needs gradle 7
-        java: [8, 11]
+        os: [ ubuntu-latest, macos-latest ]
+        java: [ 8, 11, 17, 19 ]
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-java@v3
-      with:
-        java-version: ${{ matrix.java }}
-        distribution: temurin
-        cache: gradle
-    - run: |
-        ./gradlew clean check --refresh-dependencies
-        ./gradlew clean check --refresh-dependencies -PfasterxmlVersion=2.8.11
-        ./gradlew clean check --refresh-dependencies -PfasterxmlVersion=2.9.10
-        ./gradlew clean check --refresh-dependencies -PfasterxmlVersion=2.10.4
-        ./gradlew clean check --refresh-dependencies -PfasterxmlVersion=2.11.3
-        ./gradlew clean check --refresh-dependencies -PfasterxmlVersion=2.12.3
-        ./gradlew clean check --refresh-dependencies -PfasterxmlVersion=2.13.4
-        ./gradlew clean check --refresh-dependencies -PfasterxmlVersion=2.14.1
-        ./gradlew codeCoverageReport
-  validation:
-    name: "Gradle wrapper validation"
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: ${{ matrix.java }}
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.6
+
+      - name: Execute Gradle build
+        run: ./gradlew build
```

Comment:
I am not quite sure the purpose of gradle build here. Isn't grade check include compilation (which builds everything) already?

Reply:
Yes the check is already building with various JDK versions, the build here is build on various operating systems and JDK versions also, agree that there is not much value on it, but as someone who have macOs I can see if it's builds on that operating system or not, @mincong-h do you think I should keep only one operating system and remove this step ?

Reply:
Let's keep what you currently have. I am just asking out of curiosity. Having macOS build LGTM